hack/update-codegen.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# more info: https://cloud.redhat.com/blog/kubernetes-deep-dive-code-generation-customresources
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+mkdir -p ./api/clustergroupupgradesoperator
+cd ./api/clustergroupupgradesoperator
+ln -s ../v1alpha1 ./v1alpha1
+cd ../../
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd "${SCRIPT_ROOT}"; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
+
+# TIP: append '-v 10' with the script call below for logs
+bash ${CODEGEN_PKG}/generate-groups.sh "all" \
+    "github.com/openshift-kni/cluster-group-upgrades-operator/pkg/generated" \
+    "github.com/openshift-kni/cluster-group-upgrades-operator/api" \
+    clustergroupupgradesoperator:v1alpha1 \
+    --output-base "./pkg" \
+    --go-header-file hack/boilerplate.go.txt
+
+cp -rp "./pkg/github.com/openshift-kni/cluster-group-upgrades-operator/pkg/" "./pkg/"
+
+sed -i '' "s|api/clustergroupupgradesoperator/v1alpha1|api/v1alpha1|g" $(find pkg/generated -type f)
+
+# clean up
+rm -rf ./pkg/github.com/
+rm -rf api/clustergroupupgradesoperator
```

Comment:
Consider a cleanup function in a trap on EXIT. Because of the errexit at the beginning of the script, any failure (such as the sed command above) means this cleanup doesn't occur.

Reply:
Added trap for cleanup